gk/fib.c

```diff
@@ -980,14 +981,16 @@ ether_cache_put(struct gk_fib *neigh_fib,
 			return -1;
 	}
 
+	memcpy(&addr_copy, addr, sizeof(addr_copy));
```

Comment:
This is an interesting bug about an implicit assumption that parameter `addr` is not in `eth_cache`. Reading the code carefully, I noticed that we don't need to ask for parameter `addr` at all. We can copy the needed information directly from `eth_cache->ip_addr`. Style: you can write as a simple assignment instead of having a `memcpy()`: `addr_copy = eth_cache->ip_addr;`.

Besides dropping parameter `addr`, add a comment here to explain why the copy is necessary.